build: Alpine portability issues

Currently, when building on Alpine Linux, the build might fail in the following ways:

x86_64 && aarch64

```bash
/bitcoin/depends/work/build/x86_64-pc-linux-musl/native_capnp/1.1.0-725f3670538/src/kj/mutex.c++:37:10: fatal error: linux/futex.h: No such file or directory
   37 | #include <linux/futex.h>
      |          ^~~~~~~~~~~~~~~
compilation terminated.
```
```bash
/bitcoin/depends/work/build/x86_64-pc-linux-musl/qt/6.7.3-be399392c6d/qtbase/src/corelib/io/qfilesystemengine_unix.cpp:64:12: fatal error: linux/fs.h: No such file or directory
   64 | #  include <linux/fs.h>
      |            ^~~~~~~~~~~~
compilation terminated.
```

```bash
/bitcoin/src/common/netif.cpp:16:10: fatal error: linux/rtnetlink.h: No such file or directory
   16 | #include <linux/rtnetlink.h>
      |          ^~~~~~~~~~~~~~~~~~~
```

aarch64
```bash
/bitcoin/src/crypto/sha256.cpp:19:10: fatal error: asm/hwcap.h: No such file or directory
   19 | #include <asm/hwcap.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
```

These are all due to missing`linux-headers` (which are not installed by default). They can be installed with `apk add linux-headers`.

We don't currently have any Alpine build docs, so they could be added, with a note about needing the headers being required, or, the code could be changed.

--- a/depends/README.md
+++ b/depends/README.md
@@ -53,6 +53,18 @@ To build dependencies for the current arch+OS:
 
     gmake
 
+### Alpine
+
+    apk add bash build-base cmake curl make patch
+
+Skip the following packages if you don't intend to use the GUI and will build with [`NO_QT=1`](#dependency-options):
+
+    apk add bison linux-headers samurai pkgconf python3
+
+To build dependencies for the current arch+OS:
+
+    make
+
 ## Configuring Bitcoin Core
 
 **When configuring Bitcoin Core, CMake by default will ignore the depends output.** In

--- a/doc/build-unix.md
+++ b/doc/build-unix.md
@@ -161,8 +161,7 @@ ZMQ dependencies (provides ZMQ API):
 
     apk add zeromq-dev
 
-User-Space, Statically Defined Tracing (USDT) is not directly compatible on
-Alpine as it uses libc functionality not available on musl.
+User-Space, Statically Defined Tracing (USDT) is not supported or tested on Alpine Linux at this time.
 
 GUI dependencies:
 

--- a/doc/build-unix.md
+++ b/doc/build-unix.md
@@ -139,6 +139,46 @@ The GUI will be able to encode addresses in QR codes unless this feature is expl
 
 Otherwise, if you don't need QR encoding support, use the `-DWITH_QRENCODE=OFF` option to disable this feature in order to compile the GUI.
 
+### Alpine
+
+#### Dependency Build Instructions
+
+Build requirements:
+
+    apk add build-base cmake linux-headers pkgconf python3
+
+Now, you can either build from self-compiled [depends](#dependencies) or install the required dependencies:
+
+    apk add libevent-dev boost-dev
+
+SQLite is required for the descriptor wallet:
+
+    apk add sqlite-dev
+
+To build Bitcoin Core without wallet, see [*Disable-wallet mode*](#disable-wallet-mode)
+
+ZMQ dependencies (provides ZMQ API):
+
+    apk add zeromq-dev
+
+User-Space, Statically Defined Tracing (USDT) is not directly compatible on
+Alpine as it uses libc functionality not available on musl.
+
+GUI dependencies:
+
+Bitcoin Core includes a GUI built with the cross-platform Qt Framework. To compile the GUI, we need to install
+the necessary parts of Qt, the libqrencode and pass `-DBUILD_GUI=ON`. Skip if you don't intend to use the GUI.
+
+    apk add qt6-qtbase-dev  qt6-qttools-dev
+
+For Qt 6.5 and later, the `xcb-util-cursor` package must be installed at runtime.
+
+The GUI will be able to encode addresses in QR codes unless this feature is explicitly disabled. To install libqrencode, run:
+
+    apk add libqrencode-dev
+
+Otherwise, if you don't need QR encoding support, use the `-DWITH_QRENCODE=OFF` option to disable this feature in order to compile the GUI.
+
 ## Dependencies
 
 See [dependencies.md](dependencies.md) for a complete overview, and

